service/matching/task_reader.go

```diff
@@ -298,10 +300,12 @@ func (tr *taskReader) addSingleTaskToBuffer(
 	}
 }
 
-func (tr *taskReader) persistAckLevel(ctx context.Context) error {
+func (tr *taskReader) persistAckBacklogCountLevel(ctx context.Context) error {
```

Comment:
Hmm, we could but we won't be able to access the `taskReader` (to emit the metric) and neither would we be able to access the `db` to UpdateState. 

Do you intend on passing them as parameters if we move this to ackManager? A concern over there would be from a design standpoint since this would potentially break/harm information hiding?

Reply:
Have not refactored into ack_manager for now, shall be doing this later since ack_manager itself needs a lot of refactoring. Let me know if this sounds okay.